src/qmc_states.jl

```diff
@@ -0,0 +1,248 @@
+"""
+    SingleState(hamiltonian, algorithm, v, wm, pnorm, params, id)
+
+Struct that holds a single state vector and all information needed for an independent run
+of the algorithm. Can be advanced a step forward with [`advance!`](@ref).
+
+See also [`SpectralState`](@ref), [`SpectralStrategy`](@ref),
+[`ReplicaState`](@ref), [`ReplicaStrategy`](@ref), [`replica_stats`](@ref),
+[`PMCSimulation`](@ref).
+"""
+mutable struct SingleState{H,A,V,W,SP}
+    # Future TODO: rename these fields, add interface for accessing them.
+    hamiltonian::H
+    algorithm::A
+    v::V       # vector
+    pv::V      # previous vector.
+    wm::W      # working memory. Maybe working memories could be shared among replicas?
+    shift_parameters::SP # norm, shift, time_step; is mutable
+    id::String # id is appended to column names
+end
+
+function Base.show(io::IO, r::SingleState)
+    print(
+        io,
+        "SingleState(v: ", length(r.v), "-element ", nameof(typeof(r.v)),
+        ", wm: ", length(r.wm), "-element ", nameof(typeof(r.wm)), ")"
+    )
+end
+
+"""
+    state_vectors(state)
+Returns a collection of configuration vectors from the `state`.
+
+See also [`SingleState`](@ref), [`ReplicaState`](@ref),
+[`SpectralState`](@ref), [`PMCSimulation`](@ref).
+"""
+function state_vectors(state::SingleState)
+    return SVector(state.v,)
+end
```

Comment:
Is this (and the other method for `SpectralState`) still needed? Can it be merged into `StateVectors`?